### ISSUE TITLE
Convert `to‑run.yaml` to use the new subtest format

### ISSUE THREAD

**Issue by ExE-Boss** [jsdom/jsdom#3157]
Now that <https://github.com/jsdom/jsdom/pull/3139> is implemented, `to‑run.yaml` entries with more than 1 subtest that don’t have harness errors should be migrated to the new format.

---

The `test/web‑platform‑tests/run‑single‑wpt.js` diff that I’m using to detect these tests:
```diff
 if (errors.length === 0 && expectFail) {
   reject(new Error(unexpectedPassingTestMessage));
 } else if ((errors.length === 1 && (tests.length === 1 || harnessFail)) && !expectFail) {
   reject(new Error(errors[0]));
 } else if (errors.length && !expectFail) {
   reject(new Error(`${errors.length}/${tests.length} errors in test:\n\n${errors.join("\n\n")}`));
-} else if (typeof expectFail !== "object") {
+} else if (typeof expectFail !== "object" &&
+  (errors.length === 0 || harnessFail || tests.length === errors.length || tests.length === 1)) {
   resolve();
 } else {
```

## `dont‑upstream` tests:
- [ ] <code>[custom‑elements/CustomElementRegistry‑dont‑upstream.html](https://github.com/jsdom/jsdom/blob/03c3dd72536110b525a6616f117dee33b2e4673e/test/web-platform-tests/to-upstream/custom-elements/CustomElementRegistry-dont-upstream.html)</code> (<https://github.com/jsdom/jsdom/pull/3140>)
- [ ] <code>[domparsing/DOMParser‑dont‑upstream.html](https://github.com/jsdom/jsdom/blob/03c3dd72536110b525a6616f117dee33b2e4673e/test/web-platform-tests/to-upstream/domparsing/DOMParser-dont-upstream.html)</code>
- [ ] <code>[hr‑time/performance‑tojson‑dont‑upstream.html](https://github.com/jsdom/jsdom/blob/03c3dd72536110b525a6616f117dee33b2e4673e/test/web-platform-tests/to-upstream/hr-time/performance-tojson-dont-upstream.html)</code>
	- [ ] Depends on: <https://github.com/web-platform-tests/wpt/pull/28202>
- [ ] <code>[hr‑time/timeOrigin‑dont‑upstream.html](https://github.com/jsdom/jsdom/blob/03c3dd72536110b525a6616f117dee33b2e4673e/test/web-platform-tests/to-upstream/hr-time/timeOrigin-dont-upstream.html)</code>
- [ ] <code>[html/browsers/windows/nested‑browsing‑contexts/frameElement‑dont‑upstream.html](https://github.com/jsdom/jsdom/blob/03c3dd72536110b525a6616f117dee33b2e4673e/test/web-platform-tests/to-upstream/html/browsers/windows/nested-browsing-contexts/frameElement-dont-upstream.html)</code>
- [ ] <code>[html/semantics/forms/the‑fieldset‑element/HTMLFieldSetElement‑dont‑upstream.html](https://github.com/jsdom/jsdom/blob/03c3dd72536110b525a6616f117dee33b2e4673e/test/web-platform-tests/to-upstream/html/semantics/forms/the-fieldset-element/HTMLFieldSetElement-dont-upstream.html)</code>
- [ ] <code>[html/semantics/forms/the‑label‑element/label‑attributes‑dont‑upstream.html](https://github.com/jsdom/jsdom/blob/03c3dd72536110b525a6616f117dee33b2e4673e/test/web-platform-tests/to-upstream/html/semantics/forms/the-label-element/label-attributes-dont-upstream.html)</code>
- [ ] <code>[html/semantics/scripting‑1/the‑script‑element/script‑languages‑dont‑upstream.html](https://github.com/jsdom/jsdom/blob/03c3dd72536110b525a6616f117dee33b2e4673e/test/web-platform-tests/to-upstream/html/semantics/scripting-1/the-script-element/script-languages-dont-upstream.html)</code>
- [ ] <code>[html/semantics/scripting‑1/the‑script‑element/execution‑timing/async‑script‑depending‑on‑previous‑sync‑dont‑upstream.html](https://github.com/jsdom/jsdom/blob/03c3dd72536110b525a6616f117dee33b2e4673e/test/web-platform-tests/to-upstream/html/semantics/scripting-1/the-script-element/execution-timing/async-script-depending-on-previous-sync-dont-upstream.html)</code>

**Comment by domenic** [comment]
> Now that #3139 is implemented, to‑run.yaml entries with more than 1 subtest that don’t have harness errors should be migrated to the new format.

I don't agree with that generally. Dealing with tests with more than one failure can be a maintenance burden, as they often indicate areas of the code that we don't implement where some tests pass coincidentally, or only implement a small part of.

I agree that the dont-upstream.html files should be removed. That, in my mind, is the primary value of partially-failing tests. There we've already made the decision that the test coverage provided outweighs the maintenance cost of dealing with partially-failing test files. And the new format makes that maintenance cost even less, so it's a clear win.

And in other cases, e.g. `blob/Blob-constructor.any.html` from #3163, it's worthwhile: there's a lot of good test coverage, and only two failing tests.

But in general we should not be looking to move all partially-failing tests to the failures-expected format. Doing so requires a lot of review bandwidth and won't be high priority for me. It's certainly not a goal for this project to convert all tests in such a way.

I'd encourage you to focus on to-upstream tests, and not directory-by-directory conversions.

**Comment by domenic** [comment]
I think this is generally in good shape these days, especially after recent rolls.
